File: src/main/java/org/featx/spec/exception/BaseExceptionHandler.java
package org.featx.spec.exception;

import org.featx.spec.model.BaseResponse;

/**
 * @author Excepts
 * @since 2019/10/12 10:00
 */
public class BaseExceptionHandler {

    //    @ResponseStatus(HttpStatus.BAD_REQUEST)
    //    @ExceptionHandler({MethodArgumentNotValidException.class, ConstraintViolationException.class,
    //            HttpRequestMethodNotSupportedException.class, IllegalStateException.class})
    <R> BaseResponse<R> handleConstraintViolationException(Exception e) {
        return BaseResponse.occur(BusinessError.PARAMETER_TYPE);
    }

    //    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    //    @ExceptionHandler(value = BusinessException.class)
    BaseResponse<Object> handleInvalidUser(BusinessException e) {
        return BaseResponse.occur(e.getBusinessError(), e.getExtra());
    }

}

File: src/main/kotlin/org/featx/jvmon/exception/BaseExceptionHandler.kt
package org.featx.jvmon.exception

import lombok.extern.slf4j.Slf4j
import org.featx.jvmon.model.BaseResponse

/**
 * @author Excepts
 * @since 2019/10/12 10:00
 */
@Slf4j
class BaseExceptionHandler {

    //    @ResponseStatus(HttpStatus.BAD_REQUEST)
    //    @ExceptionHandler({MethodArgumentNotValidException.class, ConstraintViolationException.class,
    //            HttpRequestMethodNotSupportedException.class, IllegalStateException.class})
    fun <T> handleConstraintViolationException(e: Exception): BaseResponse<T> {
        return BaseResponse.occur(BusinessError.PARAMETER_TYPE)
    }

    //    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    //    @ExceptionHandler(value = BusinessException.class)
    fun <T> handleInvalidUser(e: BusinessException): BaseResponse<Any?> {
        return BaseResponse.occur(e.businessError, e.extra)
    }

}